### ISSUE TITLE
Local plugins can register game actions and mutate game state in subscriptions

### ISSUE THREAD

**Issue by IntelOrca** [OpenRCT2/OpenRCT2#16705]
### Operating System

Windows 11

### OpenRCT2 build

66b402c

### Describe the issue

Local plugins should never be able to modify the game state. Although I will probably just limit to network games for now.

* `registerAction` needs to error for local plugins.
* Hooks should always have mutable set to false for local plugins.

### Steps to reproduce

Make a local plugin and use `registerAction` or modify game state in a `subscribe` callback.

**Comment by spacek531** [comment]
This does not include modifying game state through `context.executeAction` does it?

**Comment by IntelOrca** [comment]
@spacek531 not for remote plugins

**Comment by spacek531** [comment]
I should rephrease: should local plugins be allowed to modify the game state via `context.executeAction`? It is my undestanding that local plugins are allowed to use existing actions but not new ones.

**Comment by IntelOrca** [comment]
Yes, but this issue is about `registerAction`, not `executeAction`.
